### PR TITLE
drivers: display_st7789v: obtain parameters from DT, add Waveshare 1.3inch display

### PR DESCRIPTION
**Pull request by jfischer-no**
Rework ST7789V driver, obtain parameters from DT. Add Waveshare 1.3inch 240x240 IPS LCD display support. Based on #20570

### CODE REVIEW

**Comment by jfischer-phytec-iot** [comment]
@vanwinkeljan :smile: feel free to pick it to #20570

File: boards/shields/st7789v_generic/doc/index.rst [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+   :goals: build
+
+References
+**********
+
+.. target-notes::
+
+.. _ST7789V Datasheet:
+   https://www.newhavendisplay.com/appnotes/datasheets/LCDs/ST7789V.pdf
+
```
Comment (dbkinder) [inline boards/shields/st7789v_generic/doc/index.rst]: Delete extra blank line at the end of the file

**Review by dbkinder — CHANGES_REQUESTED** [review]

**Comment by vanwinkeljan** [comment]
@jfischer-phytec-iot merged this PR with #20570, thanks for the contribution!

File: boards/shields/st7789v_generic/doc/index.rst (hunk 1 above)
Comment (vanwinkeljan) [reply boards/shields/st7789v_generic/doc/index.rst]: fixed in #20570

**Pull request closed without merging**